### ISSUE TITLE
Missing allocations 

### ISSUE THREAD

**Issue by Boquete** [mirumee/saleor#5334]
### What I'm trying to achieve
It's hard to diagnose what the problem might be, but for example:
we have 100 orders placed, which means there should be 100 allocations. But once in a while, there is missing one or two allocation. 

I can't really find what can be a problem, I wrote command to fix it (for loop for each order line and fix stock allocation quantity).

My first thought was there is problem with canceling order or creating draft, as in the past - but that works fine. It looks like 1 in XX orders is placed without allocating stock quantity

### Steps to reproduce the problem
As mentioned above, hard to reproduce, placing XXX orders on production env and comparing with script result (counting Order Lines quantity for unfulfilled orders might help) 

### Screenshots

**System information**
Operating system:
Aws EC2

**Comment by maarcingebala** [comment]
Hmm, maybe there is some race condition when two users are modifying an order/product at the same time, either in the dashboard or in the storefront? Maybe we're missing some `select_for_update` somewhere. @patrys I think you were once debugging a similar issue.

**Comment by NyanKiyoshi** [comment]
I didn't find flaws in the logic for the GraphQL API of Checkout, maybe some variants have the flag `track_inventory` set to `False`? That could be a possibility.

We are talking about stocks and `quantity_allocated`, yes? Double checking. Which version/hash of saleor?

**Comment by patrys** [comment]
I think what could be happening is some race condition where the model that contains the quantity allocated field is read and then subsequently saved for an unrelated reason. Django by default saves all model fields, not just the ones that were assigned to, so if the allocation happens between the read and the write, the write would undo the allocation.

**Comment by Boquete** [comment]
> I didn't find flaws in the logic for the GraphQL API of Checkout, maybe some variants have the flag `track_inventory` set to `False`? That could be a possibility.
> 
> We are talking about stocks and `quantity_allocated`, yes? Double checking. Which version/hash of saleor?

2.9.0 (first release)

Track_inventory is not the case - this error occurs only to products with enabled tracking inventory.
Yes - it's about quantity_allocated. Usually it's too small.

Some sample:
```
Stock (1114 Piel Tetra) allocation | 8 -> 10
Stock (1129 S1) allocation | 2 -> 3
Stock (1132 Piel Flanela) allocation | 1 -> 2
Stock (1120 Ciazowy Retro) allocation | 6 -> 7
Stock (1115 Ciazowy Fashion) allocation | 4 -> 6
```

As you can see that's usually some small numbers / differences, this sample is for products which were almost sold out and fulfilled, but even if we had 200 stock of them, at the end of sale there would be only few allocatons missings.


Although - after further testing, even after we closed the shop this missing allocations appeared. My guess it's somewhere with Fulfillment logic, but still no clue what happens.
I also check quantity, and it's all right - so quantity is reduced properly, but quantity_allocated is not.

**Comment by maarcingebala** [comment]
We discussed this issue today and we're wondering if it isn't fixed on `master` after `quantity_allocated` was moved to the `Stock` model. This model is now the main source of truth about the stock quantity of a variant and should be saved only when the quantities are changing. I would wait for PR https://github.com/mirumee/saleor-dashboard/pull/390 (should be merged this week) and see how it works once the warehouses/stock UI is fully operational again.
